lib/buildsystems/autotools.rb

```diff
@@ -5,25 +5,27 @@ class Autotools < Package
   property :pre_configure_options, :configure_options
 
   def self.build
-    puts "Additional configure_options being used: #{@pre_configure_options.nil? || @pre_configure_options.empty? ? '<no pre_configure_options>' : @pre_configure_options} #{@configure_options.nil? || @configure_options.empty? ? '<no configure_options>' : @configure_options}".orange
-    # Run autoreconf if necessary
-    unless File.executable? './configure'
-      if File.executable? './autogen.sh'
-        system 'NOCONFIGURE=1 ./autogen.sh --no-configure || NOCONFIGURE=1 ./autogen.sh'
-      elsif File.executable? './bootstrap'
-        system 'NOCONFIGURE=1 ./bootstrap --no-configure || NOCONFIGURE=1 ./bootstrap'
-      else
-        system 'autoreconf -fiv'
+    unless File.file?('Makefile')
+      puts "Additional configure_options being used: #{@pre_configure_options.nil? || @pre_configure_options.empty? ? '<no pre_configure_options>' : @pre_configure_options} #{@configure_options.nil? || @configure_options.empty? ? '<no configure_options>' : @configure_options}".orange
+      # Run autoreconf if necessary
+      unless File.executable? './configure'
+        if File.executable? './autogen.sh'
+          system 'NOCONFIGURE=1 ./autogen.sh --no-configure || NOCONFIGURE=1 ./autogen.sh'
+        elsif File.executable? './bootstrap'
+          system 'NOCONFIGURE=1 ./bootstrap --no-configure || NOCONFIGURE=1 ./bootstrap'
+        else
+          system 'autoreconf -fiv'
+        end
       end
+      abort 'configure script not found!'.lightred unless File.file?('configure')
+      FileUtils.chmod('+x', 'configure')
+      if `grep -q /usr/bin/file configure`
+        puts 'Using filefix.'.orange
+        system 'filefix'
+      end
+      @mold_linker_prefix_cmd = CREW_LINKER == 'mold' ? 'mold -run ' : ''
+      system "#{@pre_configure_options} #{@mold_linker_prefix_cmd}./configure #{CREW_OPTIONS} #{@configure_options}"
     end
```

Comment:
What do you do in cases when there is no Makefile?  Run automake?

Reply:
Also, woundn't you want to run configure if the Makefile settings were wrong?  ie. `prefix = /usr`?

Reply:
Isn't that where `def self.patch` comes in?

Reply:
Generally with autotools, you don't have a Makefile until configure runs....

Reply:
Edits in `def self.patch` is not the elegant way to do it.  The whole point of `configure` is so you can control the settings without patching.

Reply:
For sure, but if the Makefile is prepopulated, that's non-standard, no? With autotools you are supposed to have a configure script to generate the Makefile.

Yes, there are cases where a Makefile is provided, but generally those are hacks..., and in those cases we can't use the buildtools/autotools mechanism anyways.

Reply:
now gated behind `CREW_CACHE_BUILD`...

Reply:
I have seen several projects that provide a pre-generated Makefile but only some of the time are the settings correct.  I hope the configure script will be able to run the first time without any messing with the default environment settings.